subprojects/hashing/src/main/java/org/gradle/internal/hash/HashCode.java

```diff
@@ -53,11 +68,13 @@ public static HashCode fromBytes(byte[] bytes) {
         return fromBytesNoCopy(bytes.clone());
```

Comment:
You could avoid copy in case of the newly specialized class

Reply:
Good point!